challenge_presentation/src/challenge_presentation/presentation.py

```diff
@@ -86,23 +90,30 @@ def execute(self, userdata=None):
-        function_list.append(partial(self.robot.leftArm._send_joint_trajectory, [[0, 0, 0, 1.7, 0, 0, 0]]))
-        function_list.append(partial(self.robot.rightArm._send_joint_trajectory, [[0, 0, 0, 1.7, 0, 0, 0]]))
-        function_list.append(partial(self.robot.leftArm.send_gripper_goal, "open"))
-        function_list.append(partial(self.robot.leftArm.send_gripper_goal, "close"))
-        function_list.append(partial(self.robot.rightArm.send_gripper_goal, "open"))
-        function_list.append(partial(self.robot.rightArm.send_gripper_goal, "close"))
-        function_list.append(partial(self.robot.leftArm.reset))
-        function_list.append(partial(self.robot.rightArm.reset))
-        function_list.append(partial(self.robot.rightArm.wait_for_motion_done))
+        if self.robot.robot_name == "amigo":
+            function_list.append(partial(self.robot.speech.speak, self.trans.TWO_ARMS, language=self.language, voice=self.voice, block=False))
+            function_list.append(partial(self.robot.speech.speak, self.trans.HUMAN_ARMS, language=self.language, voice=self.voice, block=False))
+            function_list.append(partial(self.robot.leftArm.send_joint_trajectory, "wave_front"))
+            function_list.append(partial(self.robot.speech.speak, self.trans.END_OF_ARMS, language=self.language, voice=self.voice, block=False))
+            function_list.append(partial(self.robot.speech.speak, self.trans.GRIPPERS, language=self.language, voice=self.voice, block=False))
+            function_list.append(partial(self.robot.leftArm._send_joint_trajectory, [[0, 0, 0, 1.7, 0, 0, 0]]))
+            function_list.append(partial(self.robot.rightArm._send_joint_trajectory, [[0, 0, 0, 1.7, 0, 0, 0]]))
+            function_list.append(partial(self.robot.leftArm.send_gripper_goal, "open"))
+            function_list.append(partial(self.robot.leftArm.send_gripper_goal, "close"))
+            function_list.append(partial(self.robot.rightArm.send_gripper_goal, "open"))
+            function_list.append(partial(self.robot.rightArm.send_gripper_goal, "close"))
+            function_list.append(partial(self.robot.leftArm.reset))
+            function_list.append(partial(self.robot.rightArm.reset))
+            function_list.append(partial(self.robot.rightArm.wait_for_motion_done))
+        else:
+            function_list.append(partial(self.robot.speech.speak, self.trans.NO_ARMS, language=self.language, voice=self.voice, block=False))
+
 
         # Torso
-        function_list.append(partial(self.robot.speech.speak, self.trans.TORSO, language=self.language, voice=self.voice, block=False))
+        if self.robot.robot_name == "amigo":
+            function_list.append(partial(self.robot.speech.speak, self.trans.TORSO, language=self.language, voice=self.voice, block=False))
+        else:
+            function_list.append(partial(self.robot.speech.speak, self.trans.TORSO_SERGIO, language=self.language, voice=self.voice, block=False))
```

Comment:
The 'if/else' statement makes the `TORSO_SERGIO` valid for all other robots, yet the name suggests it's only for Sergio. I'd suggest to either fix the name (but that would end up near `TORSO` which apparently is also not generic, but Amigo specific, which thus seem to lead to more `TORSO`-renaming), or insert an assert that the robot is really Sergio, so it will at least crash when you try to run this at a third robot without fixing the names.